lib/src/SlashRemover.cc

```diff
@@ -0,0 +1,107 @@
+#include "drogon/plugins/SlashRemover.h"
+#include "drogon/HttpAppFramework.h"
+#include "drogon/utils/FunctionTraits.h"
+#include <functional>
+#include <string>
+
+using namespace drogon;
+using namespace plugin;
+using std::string;
+
+#define TRAILING_SLASH_REGEX ".+\\/$"
+#define DUPLICATE_SLASH_REGEX ".*\\/{2,}.*"
+
+enum removeSlashMode : uint8_t
+{
+    trailing = 1 << 0,
+    duplicate = 1 << 1,
+    both = trailing | duplicate,
+};
+
+inline constexpr const char* regexes[] = {
```

Comment:
Inline variables are a C++17 feature. After addressing this, LGTM.

Reply:
Strangely enough, the C++14 build passed, do we need to make modifications?

Reply:
Most compilers that support higher C++ versions can compile that code while showing warnings:

https://github.com/drogonframework/drogon/actions/runs/5064599233/jobs/9092349771?pr=1614#step:12:99

It looks like other lines are also using C++17... so let's keep it as-is...